### PR TITLE
❇️ FWF-3370: [Feature] BPM  changes to support credentials from external cred manager

### PR DESCRIPTION
**Pull request by auslin-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE
https://aottech.atlassian.net/browse/FWF-3370

# Changes
BPM  changes to support credentials from external cred manager

# Screenshot
image

Tested by adding the config to a Listener
image
image

# Checklist
- [ ] Updated changelog
- [ ] Added meaningful title for pull request

### CODE REVIEW

File: forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    .opsForKeyValue(vaultPath, KeyValueBackend.KV_2)
+                    .get(vaultSecret);
+
+            if (vaultResponse != null && vaultResponse.getData() != null) {
+                LOGGER.debug("Fetching vault data from path: {} and Secret: {}", vaultPath, vaultSecret);
+                ObjectMapper objectMapper = new ObjectMapper();
+                return objectMapper.writeValueAsString(vaultResponse.getData());
+            } else {
+                LOGGER.debug("No data found for vault for path: {} and secret: {}", vaultPath, vaultSecret);
+                return "No data found for secret " + vaultSecret;
```
Comment (sumesh-aot) [inline forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java:50]: Can we return Null instead ?

File: forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public void setVaultPath(String path) {
+        vaultPath = path;
+    }
+
+    @Value("${spring.cloud.vault.secret}")
+    public void setVaultSecret(String secret) {
+        vaultSecret = secret;
+    }
+
+    public static String getSecret() {
```
Comment (sumesh-aot) [inline forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java:38]: Can this be called statically for any listener ?

File: forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java (hunk 1 above)
Comment (auslin-aot) [reply forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java]: done

File: forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java (hunk 2 above)
Comment (auslin-aot) [reply forms-flow-bpm/forms-flow-bpm-camunda/src/main/java/org/camunda/bpm/extension/commons/utils/VaultConfig.java:38]: yes. Tried in the BPMFormDataPipelineListener
image

**Review by abilpraju-aot — APPROVED** [review]

**Pull request merged**